### ISSUE TITLE
Bootstrap - visual indication of which page is currently being viewed on page menu

### ISSUE THREAD

**Issue by FayCross** [thexerteproject/xerteonlinetoolkits#1116]
Should we have a visual indication of which page in page menu list is the current page? For screen readers we should use aria-current="page" on the current page's link. 

The section menu already has a visual indicator of which is the current section but needs aria-current="true" for screen readers

**Comment by FayCross** [comment]
Looks like this has been done in 3.12 - not sure who by!

**Comment by torinfo** [comment]
@Timo did (one of our developers)
Merged in about two weeks ago

**Comment by ronm123** [comment]
@FayCross @torinfo @TimoBoer I've re-opened this because I'm a bit concerned that it doesn't pass our usual goal of not adversely impacting existing projects. I think it needs to be controlled by an optional property that is off by default or some other method that doesn't impact custom themes by default. I've only done some quick checks so far and I think it's mostly fine with the bundled themes but any theme that has a light background to the navbar and therefore dark link text for contrast is instantly broken by this change because the contrast is wrong e.g. dark green link colour on black background.
I know we've made other changes in 3.12 that require updates to custom themes but they don't really impact existing projects but this does.

**Comment by ronm123** [comment]
@FayCross well perhaps yes but my main point was if it's on by default it mustn't break existing projects. We only have a few bundled themes with Bootstrap but I've just been checking those:
default theme - black on black so doesn't show
dark black & grey - black on black so doesn't show
Experimental section pages - not relevant as replaces nav bar 
Flat Black & Green - works fine because black bg with white text
Flat Blue & Grey - works but really ought to be the hover colour rather than black
So even with the bundled themes there are some small issues and in my develop I have at least 6 custom themes where this either breaks or requires changes to the custom theme to be optimal.
We have a navigation bar optional property so perhaps we should now add 'current page' colour options to that anyway but we can't ignore the fact that institutions have custom themes and at the moment this requires changes to those themes for existing projects to work without problems.
Yes just changing border might have less impact but would need to be tested and could still have issues or not show if on by default. I haven't looked at the code yet does it have a hard coded background colour I'm guessing it does. Perhaps a border could be triggered but without a hard coded colour so that it picks up the relevant theme colours? I don't know just thinking out loud but we definitely need to change something.

**Comment by ronm123** [comment]
@FayCross just looking at this further and the relevant commits I'm not sure what is the best way to deal with this. I could easily update the custom themes I have developed or have control of but it's obviously about all the other custom themes and projects in use out there that we can't control. We can obviously tweak the bundled themes so I'm wondering if there's another check or new custom style we can add for now so that it only impacts themes that meet a detectable condition? Obviously the aria changes are fine.

**Comment by torinfo** [comment]
@FayCross @ronm123 The idea is to update the themes. We did not do that yet.

**Comment by ronm123** [comment]
@torinfo update what themes?
Bundled themes fine.
Custom themes that we have access to fine (sort of).
But this currently causes an immediate, noticeable and undesirable impact on any theme where the default styling doesn't work and that's not good at all. We need to agree a solution to that so let's discuss in our meeting later?

**Comment by FayCross** [comment]
@ronm123 just committed a possible way of doing current page highlight. Default is underlined text (seen in default theme) as this will mean that the colour will always be ok in custom themes (easier than adding border as we don't know suitable colour for border). Can override this to have background colour instead in individual themes (only added to Apereo theme so far). Only thing left to do would be add new option to xwd so you could specify active colour background colour if you have a custom colour added to project for nav bar. What do you think?

**Comment by ronm123** [comment]
@FayCross yes I think that's more predictable and less likely to have unforeseen impact. Obviously the first part of that is what I also tried by using the new style of activePage in custom.css rather than just active in bootstrap.css and as you say that means we can change it as we see fit for bundled themes and custom themes that we control. The one thing that I would add is:
`text-underline-position: under;`
I've always hated the way the default underline style works and used border-bottom styles instead but the position option is much simpler and as I understand it works with most modern browsers these days.
I've just tested that and it seems to work fine with my tests projects so shall I commit that or do you want to add that?
Yes ideally we should add the nav bar option too.
What about the other bundled themes are you going to change those or shall I? I think in most cases the activePage background colour should be the same as the hover colour do you agree?

**Comment by FayCross** [comment]
@ronm123 fine for you to change the default css for underline. Could you change the themes please? I think you made most of them so will have a better idea of colour to use. I don't mind which colour you choose.

**Comment by ronm123** [comment]
@FayCross @torinfo ok I'll add the change to custom.css and check and update the bundled themes tomorrow.

**Comment by ronm123** [comment]
@FayCross @torinfo You may have seen I committed changes to each of the bundled site themes on Friday but haven't yet closed this issue because I have a quick question about the default theme:
I see this currently has an activePage background colour of #333 on your computers is that visible enough?
On my normal monitors I would say it's ok but on my 4k monitor it's possibly a bit too subtle.
So should we make the #333 just a bit lighter e.g. #404040
Also should we add the link hover colour so #6FD9D9 and in that case the background possibly doesn't need to change?
